slowquant/qiskit_interface/interface.py

```diff
@@ -160,70 +176,124 @@ def construct_circuit(self, num_orbs: int, num_elec: tuple[int, int]) -> None:
+            # Check if circuit has been transpiled
+            # In case of switching to ISA in later workflow
+            if not self._transpiled and hasattr(self, "circuit"):
+                if self.pass_manager is None:
+                    # We have switched to ISA and had no PassManager before.
+                    # This will initialize the standard internal transpilation.
+                    self.pass_manager = None
+                elif self._internal_pm:
+                    # We have used internal PassManager before but re-transpilation was requested (probs via change_primitive in WF)
+                    self.pass_manager = None
+                self.circuit = self.circuit
+
+    @property
+    def pass_manager(self) -> None | PassManager:
+        """Get PassManager.
+
+        Returns:
+            PassManager
+        """
+        return self._pass_manager
+
+    @pass_manager.setter
+    def pass_manager(self, pass_manager: None | PassManager) -> None:
+        """Set PassManager.
+
+        Args:
+            pass_manager: PassManager object from Qiskit.
+        """
+        self._internal_pm = False
+        if pass_manager is not None and not self.ISA:
+            raise ValueError("You need to enable ISA if you want to use a custom PassManager.")
+        if pass_manager is None and self.ISA:
+            self._internal_pm = True
+            self._pass_manager: None | PassManager = generate_preset_pass_manager(
+                self._primitive_level, backend=self._primitive_backend
+            )
+            print(
+                f"You selected ISA but did not pass a PassManager. Standard internal transpilation will use backend {self._primitive_backend} with optimization level {self._primitive_level}"
+            )
+        else:
+            self._pass_manager = pass_manager
+
+            # Check if circuit has been set
+            # In case of switching to new PassManager in later workflow
+            if hasattr(self, "circuit"):
+                self.circuit = self.circuit
 
     def _check_layout(self, circuit: QuantumCircuit) -> None:
         """Check if transpiled layout has changed.
 
         Args:
             circuit: Circuit whose layout is to be checked.
         """
+        # legacy function. Not needed anymore with one-time transpilation.
         if self._save_layout and circuit.layout is not None:
             if self._ISA_layout is None:
                 self._ISA_layout = circuit.layout.final_index_layout()
             else:
                 if not np.array_equal(self._ISA_layout, circuit.layout.final_index_layout()):
                     print("WARNING: Transpiled layout has changed from readout error run.")
 
-    def redo_M_mitigation(self) -> None:
-        """Redo M_mitigation."""
+    def redo_M_mitigation(self, shots: int | None = None) -> None:
+        """Redo M_mitigation.
+
+        Args:
+            shots: If defined, overwrites QI internal shot number.
```

Comment:
It is always to defined, just the default is None.

I see that this goes _make_Minv() -> _one_call_sampler_distributions(), and in _one_call_sampler_distributions() shots is not overwritten if None.

Reply:
yes